02_housing/Homelessness_metrics_city.do

```diff
@@ -247,187 +224,119 @@ collapse (sum) *homeless* *black* *hispanic* *other* *white* enrollment , by(yea
+	foreach var in homeless {
 	replace `var'_count=-1  if `var'_share>1 & `var'_share!=.
 	replace `var'_count_lb=-1 if `var'_share>1 & `var'_share!=.
 	replace `var'_count_ub=-1 if `var'_share>1 & `var'_share!=.
 	replace `var'_share=-1 if `var'_share>1 & `var'_share!=.
 	replace `var'_quality=-1 if `var'_share>1 & `var'_share!=.
 	}
 	
-*new as of 3/4/2024
-		*When the subgroup summed total is <70% or >110% of total homeless, replace with -1 or N/A
-		*create a variable set to . instead of -1 so we can sum & find the share of subgroup out of total homeless
-		*we have to do this now bc we want it based on the -1s/NAs we changed in the section above
-		foreach var in black_count white_count hispanic_count other_count {
-		gen `var'_missing = `var'
-		replace `var'_missing=. if `var'==-1
-		}
-		*create variable to tell us the percent of reported race subgroups of total homeless
-		egen total_race = rowtotal(black_count_missing white_count_missing hispanic_count_missing other_count_missing)
-		gen share_race_total = (total_race/homeless_count)*100
-		replace share_race_total =. if black_count_missing==. // because of the code above, black is the same as all other race missings
-		
-		*replace all with -1/NA when they are <70% & >110%
-		foreach var in black white hispanic other {
-		replace `var'_count=-1 if (share_race_total<70 | share_race_total>110) & share_race_total!=.
-		replace `var'_count_lb=-1 if (share_race_total<70 | share_race_total>110) & share_race_total!=.
-		replace `var'_count_ub=-1 if (share_race_total<70 | share_race_total>110) & share_race_total!=.
-		replace `var'_share=-1 if (share_race_total<70 | share_race_total>110) & share_race_total!=.
-		replace `var'_quality=-1 if (share_race_total<70 | share_race_total>110) & share_race_total!=.
-		}
-
-		*for cell sizes <=2, set all variables to N/A
-		foreach var in homeless black white hispanic other {
-		replace `var'_count_lb=-1 if `var'_count<=2
-		replace `var'_count_ub=-1 if `var'_count<=2
-		replace `var'_share=-1 if `var'_count<=2
-		replace `var'_quality=-1 if `var'_count<=2
+*for cell sizes <=2, set all variables to -1 (N/A) - reapplies suppression rules from before
+	foreach var in homeless  {
+	replace `var'_count_lb=-1 if `var'_count<=2
+	replace `var'_count_ub=-1 if `var'_count<=2
+	replace `var'_share=-1 if `var'_count<=2
+	replace `var'_quality=-1 if `var'_count<=2
+	replace `var'_count=-1 if `var'_count<=2
+	}
 		
-		replace `var'_count=-1 if `var'_count<=2
-		}
+*merge to crosswalk of places/cities
+	merge 1:1 year state city_name using "intermediate/cityfile.dta"
+	tab year _merge
+		drop if _merge==1 // drop cities that don't merge to the crosswalk
+		keep if year<=2018
+		drop _merge
 		
-*check quality
-sum coverage*, d, if homeless_quality==1
-sum coverage*, d, if homeless_quality==2
-sum coverage*, d, if homeless_quality==3
-
-drop enrollment coverage* enroll_* *_districts_suppress min_* count_supp_* *_missing total_race 
+****************
+*Quality Checks
```

Comment:
Is it possible to build quality checks as pass/fail? I'm not sure if this functionality is built in to Stata or if there is a way to create a pass/fail test. I don't see any glaring issues with the check but am also not entirely sure what would be a red flag here.